karate-core/src/main/java/com/intuit/karate/http/HttpClient.java

```diff
@@ -169,7 +169,7 @@ private T buildRequestInternal(HttpRequest request, ScriptContext context) {
                 buildCookie(cookie);
             }
         }
-        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method)) {
+        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method) || "DELETE".equals(method)) {
```

Comment:
sorry to think of this a little late - maybe we can simplify the logic. so if body == null or isNull, just return null.

else, no matter what the http method is, just do the other logic. we can completely avoid any check for "POST" "PUT" etc.

can you try this, and if all tests are green - we can do it this way then.